docker/test_env/besu.go

```diff
@@ -0,0 +1,221 @@
+
+	return networkConfig, nil
+}
+
+func (g *Besu) GetInternalExecutionURL() string {
+	return g.InternalExecutionURL
+}
+
+func (g *Besu) GetExternalExecutionURL() string {
+	return g.ExternalExecutionURL
+}
+
+func (g *Besu) GetInternalHttpUrl() string {
+	return g.InternalHttpUrl
+}
+
+func (g *Besu) GetInternalWsUrl() string {
+	return g.InternalWsUrl
+}
+
+func (g *Besu) GetExternalHttpUrl() string {
+	return g.ExternalHttpUrl
+}
+
+func (g *Besu) GetExternalWsUrl() string {
+	return g.ExternalWsUrl
+}
+
+func (g *Besu) GetContainerName() string {
+	return g.ContainerName
+}
+
+func (g *Besu) GetContainer() *tc.Container {
+	return &g.Container
+}
+
+func (g *Besu) getContainerRequest(networks []string) (*tc.ContainerRequest, error) {
+	return &tc.ContainerRequest{
+		Name:     g.ContainerName,
+		Image:    g.image,
+		Networks: networks,
+		// ImagePlatform: "linux/x86_64", //don't even try this on Apple Silicon, the node won't start due to JVM error
+		ExposedPorts: []string{NatPortFormat(TX_GETH_HTTP_PORT), NatPortFormat(TX_GETH_WS_PORT), NatPortFormat(ETH2_EXECUTION_PORT)},
+		WaitingFor: tcwait.ForAll(
+			tcwait.ForLog("Ethereum main loop is up").
+				WithStartupTimeout(120 * time.Second).
+				WithPollInterval(1 * time.Second),
+		),
+		User: "0:0", //otherwise in CI we get "permission denied" error, when trying to access data from mounted volume
+		Cmd: []string{
+			"--data-path=/opt/besu/execution-data",
+			fmt.Sprintf("--genesis-file=%s/besu.json", GENERATED_DATA_DIR_INSIDE_CONTAINER),
+			fmt.Sprintf("--network-id=%d", g.chainConfg.ChainID),
+			"--host-allowlist=*",
+			"--rpc-http-enabled=true",
+			"--rpc-http-host=0.0.0.0",
+			fmt.Sprintf("--rpc-http-port=%s", TX_GETH_HTTP_PORT),
+			"--rpc-http-api=ADMIN,CLIQUE,ETH,NET,DEBUG,TXPOOL,ENGINE,TRACE,WEB3",
+			"--rpc-http-cors-origins=*",
+			"--rpc-ws-enabled=true",
+			"--rpc-ws-host=0.0.0.0",
+			fmt.Sprintf("--rpc-ws-port=%s", TX_GETH_WS_PORT),
+			"--rpc-ws-api=ADMIN,CLIQUE,ETH,NET,DEBUG,TXPOOL,ENGINE,TRACE,WEB3",
+			"--engine-rpc-enabled=true",
+			fmt.Sprintf("--engine-jwt-secret=%s", JWT_SECRET_FILE_LOCATION_INSIDE_CONTAINER),
+			"--engine-host-allowlist=*",
+			fmt.Sprintf("--engine-rpc-port=%s", ETH2_EXECUTION_PORT),
+			"--sync-mode=FULL",
+			"--data-storage-format=BONSAI",
+			// "--logging=DEBUG",
+			"--rpc-tx-feecap=0",
+		},
+		Env: map[string]string{
+			"JAVA_OPTS": "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n",
+		},
+		Mounts: tc.ContainerMounts{
+			tc.ContainerMount{
+				Source: tc.GenericBindMountSource{
+					HostPath: g.generatedDataHostDir,
+				},
+				Target: tc.ContainerMountTarget(GENERATED_DATA_DIR_INSIDE_CONTAINER),
+			},
+		},
+	}, nil
+}
+
+func (g *Besu) WaitUntilChainIsReady(waitTime time.Duration) error {
+	waitForFirstBlock := tcwait.NewLogStrategy("Imported #1").WithPollInterval(1 * time.Second).WithStartupTimeout(waitTime)
+	return waitForFirstBlock.WaitUntilReady(context.Background(), *g.GetContainer())
```

Comment:
@Tofel shouldn't `testcontext.Get` be used here?

Reply:
that was my question in priv to @tateexon actually, I think it should. Changed it and also changed all calls to `context.Background()` in `test_env` package to that one